x/nexus/keeper/msg_dispatcher.go

```diff
@@ -60,12 +60,15 @@ func (m Messenger) routeMsg(ctx sdk.Context, msg exported.WasmMessage) error {
 		return fmt.Errorf("recipient chain %s is not a registered chain", msg.DestinationChain)
 	}
 
```

Comment:
are we validating WasmMessage somewhere in core? e.g. ensure hash length is correct etc, strings don't have `_` etc.

Reply:
We are not. The ValidateBasic function just returns nil, and has a comment explaining why: https://github.com/axelarnetwork/axelar-core/blob/c7afbd0703a1f9ee1af415490ba2e931fc2bc1f7/x/nexus/exported/types.go#L360
```
	// This message can only be sent by the cosmwasm router as part of the amplifier integration into core,
	// so there is no point in validating the message here.
```
I am not sure that it makes sense to check here, because the messages have already been verified and routed and passed all of the checks in amplifier. What would it even mean for a message to fail here, since it's already been verified successfully?

Reply:
WasmMessage is always either converted to a GeneralMessage, or created from a GeneralMessage, and we validate the GeneralMessage in both flows

Reply:
I added some extra checks anyways

Reply:
Yeah, although I'm not sure if Amplifier makes certain validation checks in the same way (such as preventing `_` from existing in address strings). And I'd like to avoid relying on Amplifier to have made the checks required by Core for safety.